src/renderer/services/dex.js

```diff
@@ -222,14 +234,26 @@ export default {
         time: barPointer,
       };
 
-      bucket = apiBucketsIndex < tradeHistory.apiBuckets.length ? tradeHistory.apiBuckets[apiBucketsIndex] : null;
+      bucket = apiBucketsIndex < apiBuckets.length ? apiBuckets[apiBucketsIndex] : null;
       trade = tradesIndex < trades.length ? trades[tradesIndex] : null;
       while (trade && trade.tradeTime < from) {
         tradesIndex += 1;
         trade = tradesIndex < trades.length ? trades[tradesIndex] : null;
       }
 
-      if (bucket && bucket.time * 1000 === barPointer) {
+      if (tradesIndex >= trades.length && trade === null) {
+        break;
```

Comment:
Couldn't it be possible that there is still more bucket data that could be used even though all the trades have been iterated through? In that case shouldn't it not break the loop yet? We can discuss offline.

Reply:
nice fix